src/client/Microsoft.Rest.ClientRuntime/ServiceClient.cs

```diff
@@ -339,20 +390,27 @@ public bool SetUserAgent(string productName, string version)
         {
             if (!_disposed && HttpClient != null)
             {
-                // Clear the old user agent
-                HttpClient.DefaultRequestHeaders.UserAgent.Clear();
-                HttpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue(FXVERSION, FrameworkVersion));
-#if net45
-                HttpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue(OSINFO, OsInfo));
-#endif
+                SetDefaultUserAgentInfo();
                 HttpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue(productName, version));
-
-                // Returns true if the user agent was set 
                 return true;
             }
 
             // Returns false if the HttpClient was disposed before invoking the method
             return false;
         }
+
+        /// <summary>
+        /// Set Default information in User Agent
+        /// </summary>
+        /// <returns></returns>
+        private void SetDefaultUserAgentInfo()
+        {
+            HttpClient.DefaultRequestHeaders.UserAgent.Clear();
+            HttpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue(FXVERSION, FrameworkVersion));
+#if net45
+            // If you want to log ProductName in userAgent, it has to be without spaces
+            HttpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue(OsName, OsVersion));
```

Comment:
Would it be more consistent to always add an OsInfo header and if it's not net45 just say "unkown" or something?